src/web/mjs/connectors/WanPaMan.mjs

```diff
@@ -0,0 +1,81 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class OnePunchMan extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'wanpaman';
+        super.label = 'Wan-Pa Man';
+        this.tags = [ 'manga', 'japanese'];
+        this.url = 'http://galaxyheavyblow.web.fc2.com';
+    }
+
+    async _getMangas() {
+        return [
+            {
+                id: this.url,
+                title: this.label
+            }
+        ];
+    }
+
+    async _getChapterList(manga, callback) {
+        let request = new Request(new URL(manga.id), this.requestOptions);
+
+        Engine.Request.fetchUI( request, 'new XMLSerializer().serializeToString(document);' )
```

Comment:
Always return a promise in the injected script, so error handling is assured.
Make better use of fetchUI to perform the logic (the same way you would in the developer console with your browser).

```javascript
let script = new Promise(resolve => {
    let chapters = [...document.querySelectorAll('body a[href*="imageviewer"]')].map(link => {
        return {
            id: link.pathname,
            title: link.text.trim()
        };
    });
    resolve(chapters);
});

let request = ...
let data = await Engine.Request.fetchUI(request, script);
```